flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -537,6 +537,7 @@ public final void invoke() throws Exception {
 			afterInvoke();
```

Comment:
This method also disposes operators and will throw encountered (not log) errors.
So we could get a log like this:
```
closing operator x
closing operator y
Invoke Error
operator-y-dispose-error-message
```
Which is confusing to me.

Reply:
This basically says we want to log the first encountered disposal error.

In the catch block, we do an enforce one and clean up no matter what. So I guess `disposeAllOperators(true)` was intended to not throw exceptions, which I guess the proposed option 3 (logging "disposal error is not the root cause") might be a solution.

Reply:
There can be no root cause, just a normal shutdown.
I think change of the log message (comment below) will make it better.

Reply:
mark this as resolved according to offline discussion.